### PR TITLE
Proper model management via the CLI

### PR DESCRIPTION
**Pull request by XapaJIaMnu**
This pull request adds support for downloading and removing models entirely from the command line, to help users that don't have access to the GUI.

### CODE REVIEW

File: src/CommandLineIface.cpp [hunk 1]
```diff
@@ -101,8 +160,55 @@ void CommandLineIface::doTranslation() {
...
+        int val = (int) (percentage * 100);
+        int lpad = (int) (percentage * PBWIDTH);
+        int rpad = PBWIDTH - lpad;
+        printf("\r%3d%% [%.*s%*s]", val, lpad, PBSTR, rpad, "");
+        fflush(stdout);
+    });
+    // Download the new model. Use eventloop again to prevent premature exit before download is finished
+    connect(&network_, &Network::downloadComplete, this, [&](QFile *file, QString filename) {
+        // We use cout here, as QTextStream out gives a warning about being lamda captured.
+        std::cout << "\nModel downloaded succesffully! You can now invoke it with -m " << modelID.toStdString() << std::endl;
```
Comment (jelmervdl) [inline src/CommandLineIface.cpp:199]: Double ff 😲

**Review by jelmervdl — COMMENTED** [review]
(Haven't tested it or anything, was just curious and noticed English)

File: src/CommandLineIface.cpp (hunk 1 above)
Comment (XapaJIaMnu) [reply src/CommandLineIface.cpp]: I have no idea what you are talking about xD

File: src/CLIParsing.h [hunk 2]
```diff
@@ -14,7 +14,10 @@ static void CLIArgumentInit(QApplication& translateLocallyApp, QCommandLineParse
...
     parser.addHelpOption();
     parser.addVersionOption();
-    parser.addOption({{"l", "list-models"}, QObject::tr("List available models.")});
+    parser.addOption({{"l", "list-models"}, QObject::tr("List locally installed models.")});
+    parser.addOption({{"a", "available-models"}, QObject::tr("Connect to the Internet and list available models. Only shows models that are NOT installed locally or have a new version available online.")});
+    parser.addOption({{"d", "download-model"}, QObject::tr("Connect to the Internet and download a model."), "output", ""});
+    parser.addOption({{"r", "remove-model"}, QObject::tr("Remove a model from the local machine. Only works for models managed with translateLocally."), "output", ""});
     parser.addOption({{"m", "model"}, QObject::tr("Select model for translation."), "model", ""});
     parser.addOption({{"i", "input"}, QObject::tr("Source translation text (or just used stdin)."), "input", ""});
     parser.addOption({{"o", "output"}, QObject::tr("Target translation output (or just used stdout)."), "output", ""});
```
Comment (jelmervdl) [inline src/CLIParsing.h:23]: I'm a bit confused about these descriptions? Is the argument you pass it used as text, or is it supposed to be a file? From the code I gather it's the latter, but description is a bit unclear.

**Review by jelmervdl — APPROVED** [review]
Other than the input/output descriptions in --help, this all seems to work quite nicely for me! (Tested it using the macOS binary that's inside the .app bundle.)

**Comment by XapaJIaMnu** [comment]
I have no idea why MacOs is suddenly failing but I guess the other PR will solve this?

**Pull request merged**